// secretary/lib/engine.js
/*
    engine.js
*/

/*
    REQUIRES
 */

var cli = require('cli-color');
var util = require('util');

/*
    CONSTRUCTOR
 */

function Engine()
{
    // Default properties

    this.minFlag = 1;
    this.maxFlag = 10;

	this.minRunningFlag = this.minFlag;
	this.maxRunningFlag = this.maxFlag;
}

/*
	PUBLIC METHODS
 */

Engine.prototype.process = function(args)
{
	var string = args[0];

	// If there is data to attempt to format
	if(args.length > 1)
	{
		string = util.format.apply(util, args);
	}

	return string;
};

Engine.prototype.isValidFlag = function(flag)
{
	return (flag >= this.minFlag && flag <= this.maxFlag);
};

Engine.prototype.setMinFlag = function(flag)
{
	// Flags equaling zero pose issues for flag checking
	if(flag > 0)
	{
		this.minFlag = flag;
	}

	// The running range cannot be lower than the minimum flag
	if(this.minRunningFlag < this.minFlag)
	{
		this.minRunningFlag = this.minFlag;
	}

	if(this.maxRunningFlag < this.minFlag)
	{
		this.maxRunningFlag = this.minFlag;
	}
};

Engine.prototype.setMaxFlag = function(flag)
{
	// Flags equaling zero pose issues for flag checking
	if(flag > 0)
	{
		this.maxFlag = flag;
	}

	// The running range cannot be higher than the maximum flag
	if(this.maxRunningFlag > this.maxFlag)
	{
		this.maxRunningFlag = this.maxFlag;
	}

	if(this.minRunningFlag > this.maxFlag)
	{
		this.minRunningFlag = this.maxFlag;
	}
};

/*
	EXPORTS
*/

exports.Engine = Engine;
